src/Vm.sol

```diff
@@ -382,4 +378,8 @@ interface Vm is VmSafe {
     function transact(bytes32 txHash) external;
     // Fetches the given transaction from the given fork and executes it on the current state
     function transact(uint256 forkId, bytes32 txHash) external;
+    // Pauses gas metering (i.e. gas usage is not counted). Noop if already paused.
+    function pauseGasMetering() external;
+    // Resumes gas metering (i.e. gas usage is counted again). Noop if already on.
+    function resumeGasMetering() external;
```

Comment:
I think you currently have `noGasMetering` modifier in `StdCheatsSafe` but this in `Vm` and not `VmSafe`?

Reply:
Correct.

Reply:
That seems contradictory since the modifier uses the cheats—how can the modifier be safe, but not the cheats? I'd say put them both in safe, or both unsafe (I lean towards safe, but can see the arg for unsafe so am indifferent)

Reply:
We're already doing it - for example, `stdStorageSafe` uses `Vm` to modify state, but it is `Safe` because we are ensuring that it is safe.

Reply:
In that case we're using unsafe cheat codes in a safe way. In this case the modifier just uses the cheats in the same way you'd use them directly

Reply:
I'll move them to `VmSafe`.